src/ipc/totp-get.ts

```diff
@@ -0,0 +1,38 @@
+import makeDebug from 'debug'
+import { IpcMain, IpcMainInvokeEvent } from 'electron'
+import exec from '../utils/exec'
+
+export const channel = 'totpGet'
+
+const debug = makeDebug('kiosk-browser:totp')
+
+export interface TotpInfo {
+  isoDatetime: string
+  code: string
+}
+
+async function totpGet(): Promise<TotpInfo | undefined> {
+  try {
+    const { stdout, stderr } = await exec('sudo', [
+      '-n',
+      '/usr/local/bin/tpm2-totp',
+      'show',
+    ])
+
+    if (stderr) return undefined
+
+    const [timestamp, code] = stdout.split(': ')
+
+    return {
+      isoDatetime: new Date(timestamp).toISOString(),
+      code,
+    }
+  } catch (err) {
+    debug('could not call tpm2-totp', err)
```

Comment:
`tpm2-totp: %s', err)`

Reply:
this worked fine fwiw, but happy to change it :)